### PR TITLE
MacOS - ScrollViewer, ItemsControl, LinearGradientBrush support, misc fixes

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
I have a problem with the `ScrollViewer`'s Background. For example when the Page Content is set as follows:

    Content = new ScrollViewer()
 	{
		new Button(){Content="Test"}
	};

I get the following result (with Light theme):

image

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+					availableSizeForChild.Height -= (nfloat)verticalMargin;
+				}
+
+#if __IOS__
+				_measuredSize = _content.SizeThatFits(availableSizeForChild);
+#else
+				_measuredSize = (_content as FrameworkElement)?.SizeThatFits(availableSizeForChild) ?? default(CGSize);
+#endif
+				_measuredSize.Width += (nfloat)horizontalMargin;
+				_measuredSize.Height += (nfloat)verticalMargin;
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs]: ### [Codacy](https://app.codacy.com/gh/unoplatform/uno/commit?cid=437596529) detected an issue:
#### Message: `'_measuredSize' is null on at least one execution path.`

    
#### Currently on:
+ **Commit**:  934d3d5ec6c416105400c8b388d95495d64afa33
+ **File**:    [src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs](https://github.com/unoplatform/uno/blob/934d3d5ec6c416105400c8b388d95495d64afa33/src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs)
+ **LineNum**: [64](https://github.com/unoplatform/uno/blob/934d3d5ec6c416105400c8b388d95495d64afa33/src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.iOSmacOS.cs#L64)

**Comment by davidjohnoliver** [comment]
@MartinZikmund Hmm, I don't have that problem on my end, but it wouldn't surprise me in the least if there are bugs to iron out.

**Pull request merged**